lib/cocoapods/installer/analyzer.rb

```diff
@@ -283,11 +287,14 @@ def copy_embedded_target_pod_targets_to_host(aggregate_target, embedded_aggregat
             host_target_uuids = Set.new(aggregate_target.user_project.host_targets_for_embedded_target(embedded_user_target).map(&:uuid))
             !aggregate_user_target_uuids.intersection(host_target_uuids).empty?
           end
-          # This embedded target is hosted by the aggregate target's user_target; copy over the non-duplicate pod_targets
-          aggregate_target.pod_targets = aggregate_target.pod_targets + embedded_aggregate_target.pod_targets.select do |pod_target|
```

Comment:
no more altering `pod_targets` array *after* we created the aggregate target! 

Maybe now I can make all properties of `aggregate_target.rb` as `freeze`? :D in a follow up